src/components/scenes/CreateWalletImportScene.tsx

```diff
@@ -1,167 +1,172 @@
-                  <PrimaryButton.Text>{s.strings.string_next_capitalized}</PrimaryButton.Text>
-                )}
-              </PrimaryButton>
-            </View>
-          </View>
-        </View>
-      </SafeAreaView>
+    const pluginIds = newWalletItems.map(item => item.pluginId)
+
+    // Loop over plugin importPrivateKey
+    const promises = pluginIds.map(
+      async pluginId =>
+        await currencyConfig[pluginId].importKey(cleanImportText).catch(e => {
+          console.warn('importKey failed', e)
+        })
     )
-  }
+
+    const results = await Promise.all(promises)
+
+    const successMap: { [pluginId: string]: JsonObject } = {}
+
+    for (const [i, keys] of results.entries()) {
+      if (typeof keys === 'object') {
+        // Success
+        successMap[pluginIds[i]] = keys
+      }
+    }
+
+    // Split up the original list of create items into success and failure lists
+    const failureItems: WalletCreateItem[] = []
+    const successItems: WalletCreateItem[] = []
+
+    for (const item of createWalletList) {
+      if (successMap[item.pluginId] != null) {
+        // Any asset associated to this pluginId is good to go
+        successItems.push(item)
+      } else if (item.createWalletIds != null && item.createWalletIds[0] === PLACEHOLDER_WALLET_ID) {
+        // Token items to be enabled on existing wallets and aren't dependent on a failed import are are good to go, too
+        successItems.push(item)
+      } else {
+        // No good
+        failureItems.push(item)
+      }
+    }
+
+    if (successItems.length === 0) {
+      await Airship.show<'edit' | undefined>(bridge => (
+        <ButtonsModal
+          bridge={bridge}
+          title={s.strings.create_wallet_failed_import_header}
+          message={s.strings.create_wallet_all_failed}
+          buttons={{
+            edit: { label: s.strings.create_wallet_edit }
+          }}
+        />
+      ))
+
+      return
+    }
+
+    if (failureItems.length > 0) {
+      // Show modal with errors
+      const displayNames = failureItems.map(item => item.displayName).join(', ')
+      const resolveValue = await Airship.show<'continue' | 'edit' | 'cancel' | undefined>(bridge => (
+        <ButtonsModal
+          bridge={bridge}
+          title={s.strings.create_wallet_failed_import_header}
+          message={sprintf(s.strings.create_wallet_some_failed, displayNames)}
+          buttons={{
+            continue: { label: s.strings.legacy_address_modal_continue },
+            cancel: { label: s.strings.string_cancel_cap }
+          }}
+        />
+      ))
+
+      if (resolveValue === 'cancel' || resolveValue == null) {
+        return
+      }
+    }
+
+    navigation.navigate('createWalletCompletion', { createWalletList, walletNames, fiatCode, importText: cleanImportText })
+  })
+
+  // Scale the icon to match the height of the first MainButton container for consistency
+  const [iconContainerLayout, setIconContainerLayout] = useLayout()
```

Comment:
We don't have the linting rule switched on officially yet, but `onX` props need to have a callback whose name starts with `handle`. So `setIconContainerLayout` needs to be `handleIconContainerLayout`